articles/understanding-abstraction-in-c++/index.md

```diff
@@ -0,0 +1,191 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/understanding-abstraction-in-c++/
+title: Understanding Abstraction in C++
+description: This article will go through an introduction to abstraction along with types and how data abstraction is achieved in C++. Abstraction is the act of representing essential features without including the background details.
+author: dawe-daniel
+date: 2021-04-26T00:00:00-18:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/understanding-abstraction-in-c++/hero.jpg
+    alt: C++ Abstraction Image Example
+---
+### Introduction
+Abstraction is the act of representing essential features without including the background details. `Data abstraction` refers to providing only essential information about the data to the outside world while hiding the background details or explanations. In C++, classes provide a great level of data abstraction. They provide enough public methods to the outside world to play with the functionality of the object and to manipulate object data.
+<!--more-->
+For example, a state without actually knowing how the class has been implemented internally. For example, the `pow()` function is used to calculate the power of a number without knowing the algorithm the function follows. In fact, the underlying implementation of the sorting functionality could change between releases of the library. And as long as the interface stays the same, your function call will still work. This article will go through an introduction to abstraction along with types and how data abstraction is achieved in C++.
+
+### Prerequisites
+To follow this article, you’ll need to have:
+
+- [Codeblocks IDE]() to run the code.
```

Comment:
Almost forgot. Please add this link.

Reply:
Hi, @kayere I've added the link, thanks for the review.